src/modules/payload_deliverer/payload_deliverer.cpp

```diff
@@ -48,47 +48,50 @@ bool PayloadDeliverer::init()
 		return false;
 	}
 
-	initialize_gripper();
+	configure_gripper();
 	return true;
 }
 
-bool PayloadDeliverer::initialize_gripper()
+void PayloadDeliverer::configure_gripper()
 {
-	// If gripper instance is invalid, try initializing it
+	// If gripper instance is valid, but user disabled the gripper, de-initialize the gripper
+	if (_gripper.is_valid() && !_param_gripper_enable.get()) {
+		_gripper.deinit();
+		return;
```

Comment:
As the param requires a reboot, this is IMO not strictly necessary

Reply:
Agreed!